django/db/models/functions/base.py

```diff
@@ -203,8 +203,7 @@ def __init__(self, expression, substring, **extra):
         """
         if not hasattr(substring, 'resolve_expression'):
             substring = Value(substring)
```

Comment:
Seems to be jsut because all the functions are imported at startup, before settings are defined:

```
Traceback (most recent call last):
  File "runtests.py", line 16, in <module>
    from django.test import TestCase, TransactionTestCase
  File "/Users/adam/Documents/Projects/django/django/test/__init__.py", line 3, in <module>
    from django.test.client import Client, RequestFactory
  File "/Users/adam/Documents/Projects/django/django/test/client.py", line 12, in <module>
    from django.core.handlers.base import BaseHandler
  File "/Users/adam/Documents/Projects/django/django/core/handlers/base.py", line 7, in <module>
    from django.urls import get_resolver, set_urlconf
  File "/Users/adam/Documents/Projects/django/django/urls/__init__.py", line 1, in <module>
    from .base import (
  File "/Users/adam/Documents/Projects/django/django/urls/base.py", line 8, in <module>
    from .exceptions import NoReverseMatch, Resolver404
  File "/Users/adam/Documents/Projects/django/django/urls/exceptions.py", line 1, in <module>
    from django.http import Http404
  File "/Users/adam/Documents/Projects/django/django/http/__init__.py", line 5, in <module>
    from django.http.response import (
  File "/Users/adam/Documents/Projects/django/django/http/response.py", line 13, in <module>
    from django.core.serializers.json import DjangoJSONEncoder
  File "/Users/adam/Documents/Projects/django/django/core/serializers/__init__.py", line 23, in <module>
    from django.core.serializers.base import SerializerDoesNotExist
  File "/Users/adam/Documents/Projects/django/django/core/serializers/base.py", line 6, in <module>
    from django.db import models
  File "/Users/adam/Documents/Projects/django/django/db/models/__init__.py", line 19, in <module>
    from django.db.models.manager import Manager
  File "/Users/adam/Documents/Projects/django/django/db/models/manager.py", line 6, in <module>
    from django.db.models.query import QuerySet
  File "/Users/adam/Documents/Projects/django/django/db/models/query.py", line 21, in <module>
    from django.db.models.functions import Trunc
  File "/Users/adam/Documents/Projects/django/django/db/models/functions/__init__.py", line 1, in <module>
    from .base import (
  File "/Users/adam/Documents/Projects/django/django/db/models/functions/base.py", line 190, in <module>
    class StrIndex(Func):
  File "/Users/adam/Documents/Projects/django/django/db/models/functions/base.py", line 199, in StrIndex
    _output_field = fields.IntegerField()
  File "/Users/adam/Documents/Projects/django/django/db/models/fields/__init__.py", line 161, in __init__
    self.db_tablespace = db_tablespace or settings.DEFAULT_INDEX_TABLESPACE
  File "/Users/adam/Documents/Projects/django/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/Users/adam/Documents/Projects/django/django/conf/__init__.py", line 41, in _setup
    % (desc, ENVIRONMENT_VARIABLE))
django.core.exceptions.ImproperlyConfigured: Requested setting DEFAULT_INDEX_TABLESPACE, but settings are not configured. You must either define the environment variable DJANGO_SETTINGS_MODULE or call settings.configure() before accessing settings.
```